### PR TITLE
add prefer-const eslint rule

### PR DESCRIPTION
**Pull request by paulrosenzweig**
@kdoh & @tlrobinson, thoughts?

I found myself doing this slowly when I saw unneeded lets, so I figured I do a fix and add the rule.

I also added an `yarn eslint-fix` script, but I'm not sure if it's needed. Previosly I'd only run prettier from eslint. It seems like you can do the reverse and get prettier to run eslint, but I figured it's easiest to just add another command.

### CODE REVIEW

**Review by kdoh — APPROVED** [review]
Definitely down. 👍 I try to do this whenever I see it but way better to just do it in one swoop and enforce it.

**Comment by tlrobinson** [comment]
Works for me. Are there any other things we should fix now so we don't have to keep dealing with merge conflicts?

**Comment by paulrosenzweig** [comment]
@tlrobinson good thought!

I just scanned [this list](https://eslint.org/docs/rules/) and pull some candidates that called to me. I didn't look at the recommended ones since we have those on already.

* https://eslint.org/docs/rules/sort-imports
* https://eslint.org/docs/rules/require-await
* https://eslint.org/docs/rules/sort-keys – possibly a contentious one
* https://eslint.org/docs/rules/no-var – maybe no diff here?
* https://eslint.org/docs/rules/prefer-destructuring – this might be a bit much, but I would like to avoid the microdecision
* https://eslint.org/docs/rules/no-duplicate-imports

Of those, only sort imports is one I've thought about before. Any other ones I should toss in?

File: .eslintrc [hunk 1]
```diff
@@ -17,7 +17,8 @@
         "react/no-find-dom-node": 0,
         "flowtype/define-flow-type": 1,
         "flowtype/use-flow-type": 1,
-        "no-color-literals": 1
+        "no-color-literals": 1,
+        "prefer-const": [2, {"destructuring": "all"}],
```
Comment (tlrobinson) [inline .eslintrc]: Could you switch this from error (2) to warning (1). We fail both in CI, but only error during builds, which can be annoying for things that aren't critical errors.

Or maybe we should treat all errors as warnings in `build-watch` and `build-hot`?

**Comment by tlrobinson** [comment]
- https://eslint.org/docs/rules/sort-imports - I personally like grouping imports by functionality/directory as I outlined here https://github.com/metabase/metabase/wiki/Frontend-Style-Guide#additional-javascript-conventions but we're not very strict about it and I'm open to discussing
- https://eslint.org/docs/rules/require-await - I could go either way on this
- https://eslint.org/docs/rules/sort-keys – soft 👎 I like prioritizing / sorting by functionality (e.x. `id` always first, things like `name` etc.). I don't think object literals are usually big enough for it to be difficult to find a particular key, but i'm willing to be convinced
- https://eslint.org/docs/rules/no-var – 👍 
- https://eslint.org/docs/rules/prefer-destructuring – 🤷‍♂ might be a fairly big diff
- https://eslint.org/docs/rules/no-duplicate-imports 👍  I think this is already enforced?

**Comment by paulrosenzweig** [comment]
* `no-var` turns out we already have this on!
* `prefer-destructuring` I tried it out and don't love it. Also, the autofixing doesn't seem to handle cases it should.

So maybe, we should just merge `prefer-const` without additional ones?

**Review by tlrobinson — APPROVED** [review]
Looks good

**Pull request merged**